Source/ButtonContext.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package cs505.group1.state;
+
+/**
+ * @author CS505 Fall17 Group1
+ * @version 2017.10.29
+ */
+abstract class ButtonContext {
+    
+    private ButtonState buttonState;
+    
+    /**
+     * Constructor.  In concrete extensions buttonState  
+     * must be assigned an initial value such as
+     * this.buttonState =  new StateA();
+     */
+    public ButtonContext(){
+        this.buttonState = null;
```

Comment:
Seeing as the context doesn't really function unless it has an initial state I would make this constructor protected and require an initial state to be passed in for it's constructor.

Reply:
Changed as suggested.